diagnostic.md

```diff
@@ -6,70 +6,92 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+Rails.application.routes.draw do
+  resources :books, only: [:show]
+
+  post '/sign-up' => 'users#signup'
+  post '/sign-in' => 'users#signin'
+  delete '/sign-out/:id' => 'users#signout'
+  patch '/change-password/:id' => 'users#changepw'
+  resources :users, only: [:index, :show]
+end
+
+or
+
+Rails.application.routes.draw do
+  resources :books, except: [:new, :edit, :create, :destroy, :update, :index]
+
+  post '/sign-up' => 'users#signup'
+  post '/sign-in' => 'users#signin'
+  delete '/sign-out/:id' => 'users#signout'
+  patch '/change-password/:id' => 'users#changepw'
+  resources :users, only: [:index, :show]
+end
+
+But that latter one is kind of excessive.
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+config/routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+It creates a migration for your database with correct labelling (e.g. CreateBooks instead of CreateBook), and the migration lets you make changes to your database.
```

Comment:
It doesn't create migrations for you, but executes all of the database migration files in `db/migrate` and generates a new database schema file, `db/schema.rb`.